GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -570,21 +571,31 @@ - (void)authenticateInteractivelyWithOptions:(GIDSignInInternalOptions *)options
 
 #if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
   [additionalParameters addEntriesFromDictionary:
-      [GIDEMMSupport parametersWithParameters:options.extraParams
-                                   emmSupport:emmSupport
-                       isPasscodeInfoRequired:NO]];
+  [GIDEMMSupport parametersWithParameters:options.extraParams
+                               emmSupport:emmSupport
+                   isPasscodeInfoRequired:NO]];
 #elif TARGET_OS_OSX || TARGET_OS_MACCATALYST
   [additionalParameters addEntriesFromDictionary:options.extraParams];
 #endif // TARGET_OS_OSX || TARGET_OS_MACCATALYST
   additionalParameters[kSDKVersionLoggingParameter] = GIDVersion();
   additionalParameters[kEnvironmentLoggingParameter] = GIDEnvironment();
 
+  NSString *codeVerifier = [OIDAuthorizationRequest generateCodeVerifier];
```

Comment:
This part is a little tricky for me because it looks like GSI is taking on some of the implementation details within AppAuth-iOS. IIUC, we have to create `codeVerifier`, `codeChallenge`, and the `nonce` because we don't have an initializer that we can easily call on [OIDAuthorizationRequest](https://github.com/openid/AppAuth-iOS/blob/master/Source/AppAuthCore/OIDAuthorizationRequest.h#L211-L213) that takes only the `nonce`.

Ideally, we would have an initializer on `OIDAuthorizationRequest` that only adds the `nonce` parameter to the [simplest initializer](https://github.com/openid/AppAuth-iOS/blob/master/Source/AppAuthCore/OIDAuthorizationRequest.h#L154). That way, we can keep these implementation details/default values inside AppAuth.

What do you think?